### DISCUSSION TITLE
How can I force the ID to be its default value in an INSERT RLS Policy?

### DISCUSSION THREAD

**Discussion by Oreoezi** [supabase/supabase#21487] [Questions] [answered]
Let's say I have two tables. For the sake of abstraction, one will store bars and another will store the drinks served at those bars. The schema would be as follows:

bars(#id, name, owner) - owner being a FK to auth.users
drinks(#id, name, bar_id) - bar_id being a FK to bars.id

Now let's say I want the users that own bars to add drinks to their bars through a RLS INSERT Policy, but I want the id of the drinks to only be a value automatically assigned by the database, not something the user can control. 

I can probably achieve that with a trigger that forces NEW.id = DEFAULT, but doing that for every table that the user should be able to insert into would be a chore. Is there a better way of doing this?

**Comment by GaryAustin1** [comment, marked as answer]
If you use an identity (sequential incremented column) you can force it using the always clause instead of default.  https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-identity-column/

This would not work for default UUID columns.

**Reply by Oreoezi** [reply]
Thanks a lot.
